packages/resources/src/RemixSite.ts

```diff
@@ -0,0 +1,1395 @@
+    const cfDistributionProps = cdk?.distribution || {};
+
+    // Validate input
+    if (cfDistributionProps.certificate) {
+      throw new Error(
+        `Do not configure the "cfDistribution.certificate". Use the "customDomain" to configure the RemixSite domain certificate.`
+      );
+    }
+    if (cfDistributionProps.domainNames) {
+      throw new Error(
+        `Do not configure the "cfDistribution.domainNames". Use the "customDomain" to configure the RemixSite domain.`
+      );
+    }
+
+    // Build domainNames
+    const domainNames = [];
+    if (!customDomain) {
+      // no domain
+    } else if (typeof customDomain === "string") {
+      domainNames.push(customDomain);
+    } else {
+      domainNames.push(customDomain.domainName);
+    }
+
+    const origin = new origins.S3Origin(this.cdk.bucket);
+    const viewerProtocolPolicy =
+      cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS;
+
+    // Return the placeholder site if intended to be a placeholder;
+    if (this.isPlaceholder) {
+      return new cloudfront.Distribution(this, "Distribution", {
+        defaultRootObject: "index.html",
+        errorResponses: buildErrorResponsesForRedirectToIndex("index.html"),
+        domainNames,
+        certificate: this.cdk.certificate,
+        defaultBehavior: {
+          origin,
+          viewerProtocolPolicy,
+        },
+      });
+    }
+
+    // Build cache policies
+    const browserBuildCachePolicy =
+      cdk?.cachePolicies?.browserBuildCachePolicy ??
+      this.createCloudFrontBrowserBuildAssetsCachePolicy();
+    const publicCachePolicy =
+      cdk?.cachePolicies?.publicCachePolicy ??
+      this.createCloudFrontPublicCachePolicy();
+    const serverResponseCachePolicy =
+      cdk?.cachePolicies?.serverResponseCachePolicy ??
+      this.createCloudFrontServerResponseCachePolicy();
+
+    let defaultBehavior: BehaviorOptions;
+
+    if (this.serverFunction instanceof Api) {
+      const apiId = this.serverFunction.httpApiId;
+      const region = Stack.of(this).region;
+      const urlSuffix = Stack.of(this).urlSuffix;
+      const apiDomain = `${apiId}.execute-api.${region}.${urlSuffix}`;
+      defaultBehavior = {
+        viewerProtocolPolicy,
+        origin: new origins.HttpOrigin(apiDomain),
+        allowedMethods: cloudfront.AllowedMethods.ALLOW_ALL,
+        cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD_OPTIONS,
+        compress: true,
+        cachePolicy: serverResponseCachePolicy,
+        ...(cfDistributionProps.defaultBehavior || {}),
+      };
+    } else if (this.serverFunction != null) {
+      defaultBehavior = {
+        viewerProtocolPolicy,
+        origin,
+        allowedMethods: cloudfront.AllowedMethods.ALLOW_ALL,
+        cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD_OPTIONS,
+        compress: true,
+        cachePolicy: serverResponseCachePolicy,
+        ...(cfDistributionProps.defaultBehavior || {}),
+        edgeLambdas: [
```

Comment:
Are both ORIGIN_REQUEST and ORIGIN_RESPONSE required? I'm not too familiar w/ this... curious if only ORIGIN_REQUEST is sufficient.

Reply:
ORIGIN_REQUEST should be the only one specified explicitly. See https://aws.amazon.com/blogs/compute/building-server-side-rendering-for-react-in-aws-lambda/ as an example for rendering React on the edge. They only specify an ORIGIN_REQUEST event.